src/codegen/sys/lib_.rs

```diff
@@ -90,6 +92,40 @@ fn get_extern_crate_string(library: &ExternalLibrary) -> String {
     )
 }
 
+fn include_custom_modules(w: &mut Write, env: &Env) -> Result<()> {
+    let modules = try!(find_modules(env));
+    if !modules.is_empty() {
+        try!(writeln!(w, ""));
+        for module in &modules {
+            try!(writeln!(w, "mod {};", module));
+        }
+        try!(writeln!(w, ""));
+        for module in &modules {
+            try!(writeln!(w, "pub use {}::*;", module));
+        }
+    }
+
+    Ok(())
+}
+
+fn find_modules(env: &Env) -> Result<Vec<String>> {
+    let path = env.config.target_path.join("src");
+
+    let mut vec = Vec::<String>::new();
+    for entry in try!(fs::read_dir(path)) {
+        let path = try!(entry).path();
+        let ext = path.extension().unwrap();
```

Comment:
Also, can you replace all `.unwrap()` with `.expect(...)` please? We should try to keep this as a good habit.

Reply:
Updated but IMHO Problem with expect, that it need include "path" to be clear and `.unwrap_or_else(|| panic!(...))` take much space.